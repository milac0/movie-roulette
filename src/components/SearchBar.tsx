import React, { useContext } from "react";
import { css } from "emotion";
import { colors } from "../theme/theme";
import { DataContext } from "./../context/DataContext";

const searchbar = css`
  background: ${colors.primary};
  border-top: 1px solid #fff;
  color: white;
  padding: 1em 2em;

  .navbar-content {
    display: flex;
    justify-content: flex-end;
    margin: 0 auto;
    max-width: 1200px;

    button {
      background: ${colors.secondaryDark};
      border: none;
      border-radius: 5px;
      color: white;
      font-size: 0.75rem;
      font-weight: 700;
      margin: 0 0.5em;
      padding: 0.5em 1em;

      &:hover {
        background: ${colors.secondary};
      }

      &:active {
        transform: translateY(2px);
      }
    }
  }
`;

interface Props {}

const SearchBar: React.FC<Props> = () => {
  const { setFilterBy } = useContext(DataContext);
  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    setFilterBy(e.currentTarget.value);
  };
  return (
    <div className={searchbar}>
      <div className="navbar-content">
        <button onClick={handleClick} value="popular">
          Popular
        </button>
        <button onClick={handleClick} value="now_playing">
          Now Playing
        </button>
        <button onClick={handleClick} value="top_rated">
          Top Rated
        </button>
      </div>
    </div>
  );
};

export default SearchBar;
