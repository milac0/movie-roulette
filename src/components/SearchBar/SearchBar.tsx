import React, { useContext } from "react";
import { DataContext } from "../../context/DataContext";
import Search from "../Search/Search";
import styles from "./searchBar.scss";

interface Props {}

const SearchBar: React.FC<Props> = () => {
  const { setFilterBy, setQuery } = useContext(DataContext);
  const handleClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    setQuery("");
    setFilterBy(e.currentTarget.value);
  };
  return (
    <div className={styles.searchbar}>
      <div className={styles.container}>
        <Search />
        <button onClick={handleClick} value="popular">
          Popular
        </button>
        <button onClick={handleClick} value="now_playing">
          Now Playing
        </button>
        <button onClick={handleClick} value="top_rated">
          Top Rated
        </button>
        <button onClick={handleClick} value="upcoming">
          Upcoming
        </button>
      </div>
    </div>
  );
};

export default SearchBar;
