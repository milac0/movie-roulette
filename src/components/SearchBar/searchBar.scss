@import "../../theme/variables";

.searchbar {
  background: $primary;
  border-top: 1px solid #fff;
}

.container {
  display: flex;
  justify-content: center;
  padding: 1em 0;

  button {
    background: $secondaryDark;
    border: none;
    border-radius: 5px;
    color: white;
    font-size: 0.75rem;
    font-weight: 700;
    margin: 0 0.5em;
    padding: 0.5em 1em;

    &:hover {
      background: $secondary;
      cursor: pointer;
    }

    &:active {
      transform: translateY(2px);
    }
  }
}
