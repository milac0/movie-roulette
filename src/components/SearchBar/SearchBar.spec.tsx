import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from './SearchBar';

describe('SearchBar renders correctly', () => {
    test('snapshot renders', () => {
        const tree = renderer.create(<SearchBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

