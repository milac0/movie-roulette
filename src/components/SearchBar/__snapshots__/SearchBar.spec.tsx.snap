// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly snapshot renders 1`] = `
<div
  className="searchbar"
>
  <div
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        name="search"
        onChange={[Function]}
        placeholder="search for..."
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search!
      </button>
    </form>
  </div>
</div>
`;
